src/lazy/text/buffer.rs

```diff
@@ -1554,17 +1560,43 @@ impl<'top> TextBufferView<'top> {
     fn match_dot_followed_by_base_10_digits(self) -> IonMatchResult<'top> {
         recognize(preceded(
             complete_tag("."),
-            opt(Self::match_digits_after_dot),
+            opt(Self::match_zero_or_more_digits_after_dot),
         ))(self)
     }
 
     /// Like `match_digits_before_dot`, but allows leading zeros.
-    fn match_digits_after_dot(self) -> IonMatchResult<'top> {
+    fn match_one_or_more_digits_after_dot(self) -> IonMatchResult<'top> {
```

Comment:
🗺️ Fixing an incompleteness detection bug that occurs when the `e` in a `float` is the last thing in the buffer.